### DISCUSSION TITLE
lemonbar localtime

### DISCUSSION THREAD

**Discussion by 0323pin** [leftwm/leftwm#622] [Q&A] [answered]
Hi guys,

So, I've decided to give it a go with `lemonbar`, I've been running without a bar or panel on my system for over a year now.

I've slightly modified a few things to use `lemonbar-xft` and have support for ttf fonts.
Everything is running fine but, I don't seems to understand how to change the format of the date/time output. Right now it shows with the US format, _i.e._ 12/12/2021 1:20 PM. This is rather annoying for an European :)
I'd like to have it as 2021-12-12 13:20
How can this be done? I've tried to change `localtime` in `template.liquid` to other functions but without any success.
Bellow is the `template.liquid` I'm using

```
{% assign mine_open = '%{F#232323}%{B#A6BC99}' %}
{% assign mine_close = '%{B-}%{F-}' %}
{% assign visible_open = '%{F#232323}%{B#C0C5CE}' %}
{% assign visible_close = '%{B-}%{F-}' %}
{% assign busy_open = '%{F#232323}%{B#FAC863}' %}
{% assign busy_close = '%{B-}%{F-}' %}

{% for tag in workspace.tags %}
{% if tag.mine %}
{{mine_open}}  {{ tag.name }}  {{mine_close}}
{% elsif tag.visible  %}
{{visible_open}}  {{ tag.name }}  {{visible_close}}
{% elsif tag.busy  %}
{{busy_open}}  {{ tag.name }}  {{busy_close}}
{% else tag.visible  %}
  {{ tag.name }}   
{% endif %}
{% endfor %}
%{c}
{{ window_title }}
%{r}
{{ localtime }}   
```

Thanks!

**Comment by AethanFoot** [comment, marked as answer]
The format cant be changed as it is fixed in leftwm-state, however this will probably be depreciated at some point due to issues in chrono which is used for that tag (mentioned in #594). You could combine the output of leftwm-state (may be easiest to use leftwm-state -q -t ... to run once) and another way of getting time (like the example on the arch wiki https://wiki.archlinux.org/title/lemonbar#Examples).

**Reply by 0323pin** [reply]
Thanks.
Yes, I was thinking the same, doing a while loop in a shell script but, though I'd ask first.

**Reply by VuiMuich** [reply]
speaking of `date`, I would actually prefer a variant of this command, that prints with a tail, instead of having to call it regularly in order to tight on the actual time. Does anyone know anything like this?

**Reply by 0323pin** [reply]
Actually, after over a year without a bar, I've just removed `lemonbar` and went back to [citron](https://github.com/grtcdr/citron).
You just need a notification daemon and a keybinding ;)

![2021-12-12-220348_1366x768_scrot](https://user-images.githubusercontent.com/90570748/145729896-7b78957d-f0eb-4497-befa-65e40f250bb6.png)

**Reply by VuiMuich** [reply]
This is beautiful, may I ask for your macchina theme?
What notify daemon are you using?

**Reply by 0323pin** [reply]
Thanks. I'm using `lxqt-notificationd`.
As for the `macchina` theme, it has been merged into `macchina`, https://github.com/Macchina-CLI/macchina/blob/main/contrib/themes/Lithium.toml

Just be aware that I'm using v6.0.0, which isn't released yet. In other words, you might need to comment out a few lines for `Lithium.toml` to work with the released version.

**Reply by 0323pin** [reply]
@VuiMuich if you need/want something else feel free to grab it from my dots, https://codeberg.org/pin/leftwm-netbsd

**Reply by VuiMuich** [reply]
Thanks!
BTW: you might be interested in [wired-notify](https://github.com/Toqozz/wired-notify), I am currently playing with it and it's quite nice. And it's Rust.

**Reply by 0323pin** [reply]
Thanks!
This looks indeed interesting 😄 
Specially, as it doesn't seem to require `polkit` and I could get rid of `kwindowsystem` as well. Also it's using `dbus` bindings and not `zbus`, which sounds promissing.

I'll look at building and packaging this.

**Comment by di-effe** [comment]
Hi, I was doing some investigation on the liquid template and this might come handy to you
https://shopify.github.io/liquid/filters/date/

I just tested with 
`{{ "now" | date: "%Y-%m-%d %H:%M" }}`
and it works

**Reply by 0323pin** [reply]
This works perfectly, thank you.
Also the link you've sent is really useful, so thank you for that also.

**Reply by di-effe** [reply]
Together we grow :D

**Reply by VuiMuich** [reply]
is this updating as time changes, or is it only pushed, when leftwm state updates? (currently not on my machine to try this myself)

I'll have to check also if raw `leftwm state` has the `localtime` field as well, if so curious I never noticed actually.

**Reply by di-effe** [reply]
Just checked on my personal computer and looks updated (just on the wrong timezone lol).
Not sure how often it is updating because I am using another computer right now

**Reply by di-effe** [reply]
Nope, it is not updating in realtime, but during leftwm refreshes :(

**Reply by VuiMuich** [reply]
Sad, but ok, I think I need to get my 'uhrwerk' project back going..

**Reply by di-effe** [reply]
What's that about?

**Reply by VuiMuich** [reply]
It's a cli RIIR of http://ticketack.de/
I also intended to add a flag that allows to simply print formatted time to stdout so widgets like eww can simply grab the tail as I find it annoying to call `datetime` every few seconds.

**Comment by 0323pin** [comment]
This basically means it will not update if you don't open/close a client or, if you don't change workspaces. Which means _I'm idle_, in which case, I'll probably be watching the time somewhere else or, I don't mind what time it is 😄

**Reply by 0323pin** [reply]
Which reminded me, I need to take `wired-notify` for a spin.

**Reply by VuiMuich** [reply]
Changing window focus triggers it as well. Same as any other command or action that manipulates windows.

"Dont't mind time" seems like good advice 😆

**Comment by 0323pin** [comment]
@VuiMuich and @di-effe Actually, time does **not** update either when using the original `template.liquid` provided within basic-lemonbar. So right now the only difference when using `{{ "now" | date: "%Y-%m-%d %H:%M" }}` instead of `{{localtime}}` is that the former outputs UTC time.

**Reply by VuiMuich** [reply]
Yeah, code doc refers to a [bug in liquid](https://github.com/cobalt-org/liquid-rust/issues/332):
```rust
        //liquid only does time in utc. BUG: https://github.com/cobalt-org/liquid-rust/issues/332
        //as a workaround we are setting a time locally
        globals.insert(
            "localtime".into(),
            liquid::model::Value::scalar(get_localtime()),
        );
```
so I believe `{{ "localtime" | date: "%Y-%m-%d %H:%M" }}` could work as well?

**Reply by 0323pin** [reply]
This simply prints the word **localtime** to the bar. Removing the quotation marks (") around localtime give the same output as localtime alone.

I'll live with US time until I get back to a notify solution, no worries.

**Reply by 0323pin** [reply]
@VuiMuich This is how it looks for now 😄 
https://www.reddit.com/r/unixporn/comments/rhnmth/leftwm_key/

**Comment by 0323pin** [comment]
@VuiMuich I've finally found the time to package `wired-notify`, it´s working great. Thanks for the suggestion.
I've got rid of the bar and I'm now using `wired-notify` + `citron` to get the information I need on demand.

**Reply by VuiMuich** [reply]
I gratefully forward the thanks to @mautamu who recommended it to me just a few day earlier :grinning:
